logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageRepository.kt

```diff
@@ -211,6 +215,14 @@ class MessageDataSource(
         )
     }
 
+    override suspend fun persistSystemMessageToAllConversations(
+        message: Message.System
+    ): Either<CoreFailure, Unit> = wrapStorageRequest {
+        messageMapper.fromMessageToEntity(message).let {
+            if (it is MessageEntity.System) messageDAO.persistSystemMessageToAllConversations(it)
```

Comment:
Should this return an `OnlySystemMessageAllowed` in the else case?

Reply:
I added the error for it, but missed it in the final step here :D thanks for mentioning it.